src/tcgwars/logic/impl/gen7/CosmicEclipse.groovy

```diff
@@ -2225,7 +2230,7 @@ public enum CosmicEclipse implements LogicCardInfo {
             energyCost P
             onAttack {
               def list = LUtils.selectMultiPokemon(bg.oppClient(), opp.bench, "Opponent used Charming Stamp. Select a Pokémon to have 90 damage dealt to it.", 1)
-              opp.bench.findAll { !list.contains(it) }.each {
+              opp.bench.findAll { list.contains(it) }.each {
                 damage 90 it
               }
             }
```

Comment:
Couldn't this use

`damage 90, opp.all.oppChoose("Opponent used Charming Stamp. Select a Pokémon to have 90 damage dealt to it.")` 

instead? Also, as it is right now it wouldn't be able to pick the active.

Reply:
You're right. I spent way too much time trying to figure out how this even was trying to do what it was trying to do, and failing to remember what `oppSelect` was called, I missed that.